packages/cli/src/commands/deploy.ts

```diff
@@ -5,18 +5,42 @@ import { BoosterConfig, Logger } from '@boostercloud/framework-types'
 import { Script } from '../common/script'
 import Brand from '../common/brand'
 import { logger } from '../services/logger'
+import { exec } from 'child-process-promise'
+import { wrapExecError } from '../common/errors'
+import * as path from 'path'
 import { currentEnvironment, initializeEnvironment } from '../common/environment'
 
+async function pruneDependencies(config: BoosterConfig): Promise<void> {
+  try {
+    await exec('npm install --production', { cwd: projectDir(config) })
+  } catch (e) {
+    throw wrapExecError(e, 'Could not prune dev dependencies')
+  }
+}
+
+async function reinstallDependencies(skipRestoreDependencies: boolean, config: BoosterConfig): Promise<void> {
+  try {
+    if(skipRestoreDependencies === false){
+      await exec('npm install', { cwd: projectDir(config) })
+    }
+  } catch (e) {
+    throw wrapExecError(e, 'Could not reinstall dependencies')
+  }
+}
+
```

Comment:
Some notes here, if you have previous execution you will have installed dev dependencies in `node_modules` so the `npm install --production` won't delete those dependencies. That's why we need to execute the `prune` instead. But if we do that the deployment won't work because as you said @javiertoledo, we still have a provider dev dependency. Should we add the provider dependencies alongside the `cli` to avoid that?

Another option could be passing a function that will be called just in time after resolving the provider and before deployment, some kind of hook, but I'm still not sure where to execute that.

And the last option and the dirtiest one could use a different directory for dev and prod dependencies, but that involves some changes on the fly I think.